File: clients/typescript/examples/tables-insert-many-behavior.ts
import { DataAPIClient, date } from "@datastax/astra-db-ts";

// Get an existing table
const client = new DataAPIClient();
const database = client.db("**API_ENDPOINT**", {
  token: "**APPLICATION_TOKEN**",
});
const table = database.table("**TABLE_NAME**");

// Insert rows into the table
(async function () {
  const result = await table.insertMany(
    [
      {
        title: "Computed Wilderness",
        author: "Ryan Eau",
        number_of_pages: 432,
        due_date: date("2024-12-18"),
        genres: new Set(["History", "Biography"]),
      },
      {
        title: "Desert Peace",
        author: "Walter Dray",
        number_of_pages: 355,
        rating: 4.5,
      },
    ],
    {
      chunkSize: 2,
      concurrency: 2,
      ordered: false,
    },
  );
})();

File: clients/go/examples/tables-insert-many-behavior.go
package main

import (
	"context"
	"log"

	"github.com/datastax/astra-db-go/v2/astra"
	"github.com/datastax/astra-db-go/v2/astra/datatypes"
	"github.com/datastax/astra-db-go/v2/astra/options"
)

func main() {
	ctx := context.Background()
	// Get an existing table
	client := astra.NewClient()

	database := client.Database(
		"**API_ENDPOINT**",
		options.API().SetToken("**APPLICATION_TOKEN**"),
	)

	table := database.Table("**TABLE_NAME**")

	// Insert rows into the table
	_, err := table.InsertMany(
		ctx,
		[]map[string]any{
			{
				"title":           "Computed Wilderness",
				"author":          "Ryan Eau",
				"number_of_pages": 432,
				"due_date": datatypes.DateOnly{
					Year:  2024,
					Month: 12,
					Day:   18},
				"genres": []string{"History", "Biography"},
			},
			{
				"title":           "Desert Peace",
				"author":          "Walter Dray",
				"number_of_pages": 355,
				"rating":          4.5,
			},
		},
		options.TableInsertMany().
			SetChunkSize(2).
			SetConcurrency(2).
			SetOrdered(false),
	)
	if err != nil {
		log.Fatal(err)
	}
}
